app/models/katello/content_view.rb

```diff
@@ -637,6 +638,14 @@ def check_ready_to_publish!(importing: false, syncable: false)
       true
     end
 
+    def check_repositories_blocking_publish!
+      repositories_with_blocking_tasks = repositories.select { |repo| repo.blocking_tasks }
+
+      if repositories_with_blocking_tasks.any?
+        fail _("Pending tasks detected in repositories of this content view. Please wait before publishing.")
```

Comment:
Would it make sense to list the tasks which are preventing this one from running? Something like we did [on lock conflicts](https://github.com/theforeman/foreman-tasks/blob/89864f50d472ac719997b2e52eb15219ad9fcbad/app/models/foreman_tasks/lock.rb#L10)

Reply:
Updated..